00_Base/src/services/sessions.service.ts

```diff
@@ -33,8 +33,8 @@ export class SessionsService {
       this.transactionRepository.getTransactionsCount(dateFrom, dateTo),
     ]);
 
-    const sessions = await this.sessionMapper.mapTransactionsToSessions(
-      transactions,
+    const sessions = this.filterBasedOnCountryCodePartyId(
```

Comment:
Since we filter after we fetch the Sessions, wouldn't the page size potentially decrease?

Reply:
I think for pagination to really work properly we need to make sure that the DB queries are written to perform operations like identifying transactions pertaining to a given partyId / countryCode combination. In other words, the pagination limit, offset, sort, etc should all be done via SQL. Ideally, if we were to be able to query DB for all new transactions where we join the location, charging station, ocpi location, etc, tables and in one query can fetch latest set of transactions via the query then I think we will be closer to having pagination working better. Otherwise, what tends to happen is you end up getting a sub set of data and cant properly have pagination, or are forced to fetch all data and have some sort of local cache layer.

Since we are using `sequelize-typescript` you would have to follow their syntax similar to what we have here:
```
await this.transactionEventRepository.readAllTransactionsByQuery({
          where: {
            stationId: stationId,
            isActive: true,
          },
          include: [
            {
              model: Evse,
              where: {
                id: evseId,
              },
            },
          ],
        })
```
where you can see the `Evse` is joined and we query based on evse id. @danielyoo227 what are your thoughts on this?

Reply:
it is worth noting with the `include: [...]` array, you can recursively include nested models and in any level down the chain you can add a `where`

Reply:
@elliot-sabitov Yep, 100%. Similar change coming up to ensure that we filter everything at DB level for country code and party ID, along with the pagination params. It'll join the OCPI Token and OCPI Locations data to check on conutry code and party ID before returning